src/app/shared/store/user.state.ts

```diff
@@ -267,6 +274,19 @@ export class UserState {
     );
   }
 
+  @Action(GetChildrenByWorkshopId)
+  getChildrenByWorkshopId(
+    { patchState }: StateContext<UserStateModel>,
+    { payload }: GetChildrenByWorkshopId
+  ): Observable<Child[]> {
+    patchState({ isLoading: true });
+    return this.achievementsService.getChildrenByWorkshopId(payload).pipe(
+      tap((approvedChildren: Child[]) => {
+        return patchState({ approvedChildren: approvedChildren, isLoading: false });
```

Comment:
I believe if there are no children, you will receive HTTP response 204 (no content), so in this case please patch the state with { totalAmount: 0, entities: []}